drivers/include/dcf77.h

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2019 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_dcf77 DCF77 long wave receiver with 77,5 kHz
+ * @ingroup     drivers_sensors
+ * @ingroup     drivers_saul
+ * @brief       Device driver long wave receiver with 77,5 kHz
+ *
+ * This driver provides @ref drivers_saul capabilities.
+ *
+ * @{
+ *
+ * @file
+ * @brief       Interface definition for the dcf77 sensor driver
+ *
+ * @author      Michel Gerlach <michel.gerlach@haw-hamburg.de>
+ */
+
+#ifndef DCF77_H
+#define DCF77_H
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <inttypes.h>
+#include "xtimer.h"
+#include "periph/gpio.h"
+#include "time.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Possible return codes
+ */
+enum {
+    DCF77_OK        =  0,       /**< all good */
+    DCF77_NOCSUM    = -1,       /**< checksum error */
+    DCF77_TIMEOUT   = -2,
+    DCF77_INIT_ERROR= -3,       /**< communication timed out */
+};
+
+/**
+ * @brief   Configuration parameters for DCF77 devices
+ */
+typedef struct {
+    gpio_t pin;             /**< GPIO pin of the device's data pin */
+    gpio_mode_t in_mode;    /**< input pin configuration, with or without pull
+                             *   resistor */
```

Comment:
Please add a comment on when the pull-up resistor would need to be used.

Reply:
For the device in the picture you only need input, no pull. For other devices that can be different.